tests/components/co2signal/conftest.py

```diff
@@ -0,0 +1,60 @@
+"""Fixtures for Electricity maps integration tests."""
+from collections.abc import Generator
+from unittest.mock import AsyncMock, patch
+
+from aioelectricitymaps import ElectricityMaps
+import pytest
+
+from homeassistant.components.co2signal import DOMAIN
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry
+from tests.components.co2signal import VALID_RESPONSE
+
+
+@pytest.fixture(name="electricity_maps")
+def mock_electricity_maps() -> Generator[AsyncMock, None, None]:
+    """Mock the ElectricityMaps client."""
+    mock = AsyncMock(
+        __aenter__=AsyncMock(
+            return_value=AsyncMock(
+                spec=ElectricityMaps,
+                latest_carbon_intensity_by_coordinates=AsyncMock(
+                    return_value=VALID_RESPONSE
+                ),
+                latest_carbon_intensity_by_country_code=AsyncMock(
+                    return_value=VALID_RESPONSE
+                ),
+            )
+        ),
+        __aexit__=AsyncMock(return_value=None),
+    )
+
+    with patch(
+        "homeassistant.components.co2signal.ElectricityMaps",
+        return_value=mock,
+    ):
```

Comment:
Any reason not to use `autospec=True` to reduce the complexity of the above AsynMock a bit?

Reply:
No, just the lack of knowledge about it, but homewizard was a good help :)